### PR TITLE
updating eventsub to match production

### PR DESCRIPTION
**Pull request by lleadbet**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

## Problem/Feature

Updates the EventSub topics to match the new topics; addresses #68 as a result.

## Description of Changes: 

- Adds Authorization Grant as `grant`
- Adds Polls events
- Adds Predictions events
- Adds Channel Gifts events
- Adds Subscription Message event
- Adds Drops entitlements event
- Adds Subscription end event
- Updates documentation to match
- Fixed small issue with .gitignore

## Checklist

- [x] My code follows the [Contribution Guide](https://github.com/twitchdev/twitch-cli/blob/master/CONTRIBUTING.md#Requirements)
- [x] I have self-reviewed the changes being requested
- [x] I have made comments on pieces of code that may be difficult to understand for other editors
- [x] I have updated the documentation (if applicable)

### CODE REVIEW

File: docs/event.md [hunk 1]
```diff
@@ -48,18 +60,18 @@ Used to either create or send mock events for use with local webhooks testing.
...
 | `--item-id`         | `-i`      | Manually set the ID of the event payload item (for example the reward ID in redemption events or game in stream events).        | `-i 032e4a6c-4aef-11eb-a9f5-1f703d1f0b92` | N               |
 | `--item-name`       | `-n`      | Manually set the name of the event payload item (for example the reward ID in redemption events or game name in stream events). | `-n "Science & Technology"`               | N               |
 | `--cost`            | `-C`      | Amount of bits or channel points redeemed/used in the event.                                                                    | `-C 250`                                  | N               |
-| `--description`     | `-d`      | Title the stream should be updated/started with.                                                                                | `-d Awesome new title!`                   | N               |
+| `--description`     | `-d`      | Title the stream should be updated/started with. Additionally used as the category ID for Drops events.                         | `-d Awesome new title!`                   | N               |
```
Comment (Jameka) [inline docs/event.md:74]: I don't see how the category ID is used in this command. The description is used to adjust the category ID for Drops-related events?

Comment (lleadbet) [reply docs/event.md:74]: Correct. Unfortunately there's not a great identifier elsewhere to use since ItemID is used for the Drop itself.

Comment (Jameka) [reply docs/event.md:74]: I don't quite agree with overloading the description parameter here when the CLI already generates a random numerical ID for the category ID if this param is missing. It's not breaking anything, but seems unnecessary when instead there could just be a custom param for category_name with a category_id autogenerated based on the string. That's more of a feature request at this point, so it's not something I'd block the release on.

Comment (lleadbet) [reply docs/event.md:74]: Totally fair, and I can remove the overload. We're getting into the weeds of needing to create a new flag (which is fine) but all the variations of "c" are taken now, so have to make a new one (possibly "g" for game?). Let me make that edit.

File: internal/events/types/drop/drop_test.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		FromUserID: fromUser,
+		ToUserID:   toUser,
+		Transport:  models.TransportEventSub,
+		Trigger:    "drop",
+	}
+
+	r, err := Event{}.GenerateEvent(params)
+	a.Nil(err)
+
+	var body models.DropsEntitlementEventSubResponse // replace with actual value
```
Comment (Jameka) [inline internal/events/types/drop/drop_test.go:30]: Comment can be removed, no?

File: docs/event.md (hunk 1 above)
Comment (lleadbet) [reply docs/event.md]: Added a new flag to address above. :)

File: internal/events/types/drop/drop_test.go (hunk 2 above)
Comment (lleadbet) [reply internal/events/types/drop/drop_test.go]: Removed!

**Review by vpolouchkine — APPROVED** [review]

**Pull request merged**